docs/understanding-your-pipeline/canonical-event/index.md

```diff
@@ -42,280 +48,403 @@ The sections below go over the standard fields found in all Snowplow events.
+</TabItem>
+<TabItem value="db" label="Databricks">
+
+The unstructured event is not part of the standard `atomic.events` columns, it is instead a `struct` type column with the name of the schema and the major version number. The fields in this table will be determined by the JSON schema defined for the event in advance, converted to snake case. Users can query fields in the self-describing event by extracting them using 
+
+```sql
+select
+...
+my_example_unstructured_event_1.myCustomField::varchar,
+...
+from 
+    atomic.events
+```
+
+</TabItem>
+</Tabs>
+
+
 #### Contexts
 
 Contexts enable Snowplow users to define their own entities that are related to events, and fields that are related to each of those entities. For example, an online retailer may choose to define a `user` context, to store information about a particular user, which might include data points like the users Facebook ID, age, membership details etc. In addition, they may also define a `product` context, with product data e.g. SKU, name, created date, description, tags etc.
 
-An event can have any number of custom contexts attached. Each context is passed into Snowplow as a JSON. Additionally, the Snowplow Enrichment process can derive additional contexts.
+An event can have any number of custom contexts attached. Each context is passed into Snowplow as a JSON. Additionally, the [Snowplow Enrichment](/docs/enriching-your-data/index.md) process can derive additional contexts.
+
+<Tabs groupId="warehouse" queryString>
+<TabItem value="rs/pg" label="Redshift/Postgres" default>
+
+Contexts are not part of the `atomic.events` table; instead, for users running on Redshift, Snowplow will shred each context JSON into a dedicated table in the `atomic` schema, making it much more efficient for analysts to query data passed in in any one of the contexts. Those contexts can be joined back to the core `atomic.events` table by the following, which is a one-to-one join or a one-to-many join.
+
+```sql
+select 
+    ...
+from 
+    atomic.events ev
+left join 
+    atomic.my_custom_context_table ctx
+    on ctx.root_id = ev.event_id and ctx.root_tstamp = ev.collector_tstamp
+```
+
+:::note
+
+You may need to take care of duplicate events as Snowplow uses an _at least once_ method of delivery.
+
+:::
+
+</TabItem>
+<TabItem value="bq" label="BigQuery">
+
+Contexts are not part of the standard `atomic.events` columns, it is instead a `REPATED` `RECORD` type column with the name of the schema and the full version number. The fields in this table will be determined by the JSON schema defined for the event in advance, converted to snake case. Users can query a single context's fields in the context by extracting them using 
+
+```sql
+select
+...
+my_custom_context_1_0_0[SAFE_OFFSET(0)].my_custom_field,
+...
+from 
+    atomic.events
+```
+
+or they can use an [`unnest`](https://cloud.google.com/bigquery/docs/reference/standard-sql/arrays#flattening_arrays) statement to explode out the array.
```

Comment:
unnest is a function, not quite a statement.

An example would be nice, particularly with a note to `LEFT JOIN` to avoid the tendency to throw out any events that don't contain the context which is a pretty common footgun.